services/src/main/java/io/grpc/services/HealthCheckingLoadBalancerFactory.java

```diff
@@ -183,6 +183,18 @@ public void handleSubchannelState(
       }
     }
 
+    @Override
+    public void shutdown() {
+      super.shutdown();
+      for (HealthCheckState hcState : helper.hcStates) {
+        // ManagedChannel will stop calling handleSubchannelState() after shutdown() is called,
+        // which is required by LoadBalancer API semantics. We need to deliver the final SHUTDOWN
+        // signal to health checkers so that they can cancel the streams.
+        hcState.updateRawState(ConnectivityStateInfo.forNonError(SHUTDOWN));
```

Comment:
This seems like it would notify `handleSubchannelState()` of our delegate after `shutdown()`. Unfortunately, putting `super.shutdown()` after this loop would also be confusing, because normally they wouldn't see the shutdown state. Do we need a "shutdown" flag to prevent notifying delegate of these transitions? Or would we prefer a flag so that the failed RPCs don't retry?